tests/functional/test_snap_dcgm.py

```diff
@@ -0,0 +1,77 @@
+import json
+import subprocess
+import urllib.request
+from time import sleep
+
+from tenacity import retry, stop_after_delay, wait_fixed
+
+
+@retry(wait=wait_fixed(5), stop=stop_after_delay(15))
+def test_dcgm_exporter():
+    """Test of the dcgm-exporter service and its endpoint."""
+    endpoint = "http://localhost:9400/metrics"
+    dcgm_exporter_service = "dcgm.dcgm-exporter"
+
+    subprocess.run(f"sudo snap start {dcgm_exporter_service}".split(), check=True)
+
+    result = subprocess.run(
+        f"sudo snap services {dcgm_exporter_service}".split(),
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+    assert " active" in result.stdout.strip(), "dcgm-exporter service is not active"
+
+    # Check the exporter endpoint, will raise an exception if the endpoint is not reachable
+    urllib.request.urlopen(endpoint)
+
+
+def test_dcgm_nv_hostengine():
+    """Check the dcgm-nv-hostengine service."""
+    nv_hostengine_service = "dcgm.nv-hostengine"
+
+    service = subprocess.run(
+        f"snap services {nv_hostengine_service}".split(),
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+
+    assert " active" in service.stdout.strip(), "nv-hostengine service is not active"
+
+
+def test_dcgmi():
+    """Test of the dcgmi command."""
+    result = subprocess.run(
+        "dcgm.dcgmi discovery -l".split(), check=True, capture_output=True, text=True
+    )
+    assert "GPU ID" in result.stdout.strip(), "DCGMI is not working"
+
+
+def test_dcgm_port_configs():
```

Comment:
Hum, just for consistency let's change this name:
```suggestion
def test_dcgm_snap_configs():
```

As you said, the exporter is an address and not a port, so I think this naming is more precise what the test is doing

Reply:
I know, but I wanted to state that the function only tests the bind configs. If we add the metrics file configuration, it will be in a separate function that will only test the metrics configuration. Maybe something like `test_dcgm_bind_configs` will be better?